iidm/iidm-tck/src/test/java/com/powsybl/iidm/network/tck/AbstractTckSuiteExhaustivityTest.java

```diff
@@ -0,0 +1,84 @@
+package com.powsybl.iidm.network.tck;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestFactory;
+import org.junit.jupiter.params.ParameterizedTest;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+public abstract class AbstractTckSuiteExhaustivityTest {
+
+    @Test
+    void assertAllTckTestsAreInstanced() {
+        ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        String tckPackageName = "com.powsybl.iidm.network.tck";
+        List<String> tckTestClasses = getPackageClasses(classLoader, tckPackageName)
+                .filter(clazz -> containsTest(classLoader, clazz))
+                .filter(clazz -> !clazz.contains("$"))
+                .collect(toList());
```

Comment:
Sonar will prompt issues for the `.collect(toList())` usages in this method since the resulting lists are never modified.
You should use `.toList()` instead.